lib/modules/manager/nuget/util.ts

```diff
@@ -210,7 +211,14 @@ function sortPatterns(
     return -1;
   }
 
-  return a[0].localeCompare(b[0]) * -1;
+  const aPathLength = (a[0].match(/\./g) ?? []).length;
+  const bPathLength = (b[0].match(/\./g) ?? []).length;
```

Comment:
Is this a complicated way to get the string length?

Reply:
I think we simply need to trim trailing `*` and do the local compare. it seems like star is sorted after dot.